addon/components/listbox/-button.js

```diff
@@ -0,0 +1,26 @@
+import Component from '@glimmer/component';
+import { action } from '@ember/object';
+
+import { modifier } from 'ember-modifier';
+
+export default class ListboxButtonComponent extends Component {
+  registerButton = modifier((element) => {
+    this.registerButtonElement(element);
+
+    return () => {
+      this.unregisterButtonElement(element);
+    };
+  });
+
+  @action registerButtonElement(element) {
```

Comment:
do these needs to be separate functions? this could probably be cleaned up a bit if inlined into the modifier?

Reply:
No, they don't. I was experiencing some odd feedback from the test suite and found that breaking them out in this way resolved the errors. That said, like the yielding issue, those errors have gone and I've refactored as you've suggested.

Think I just had some bad luck with test feedback.

Reply:
Yeah, i think this test suite can be a bit flaky at times